src/graph.py

```diff
@@ -0,0 +1,278 @@
+import logging
+import time
+
+import ccxt
+
+log = logging.getLogger(__name__)
+
+
+class PricePath:
```

Comment:
1. What do you think about factoring the graph algorithm out? I think that it would be nice to seperate them since the graph algorithm alone does a huge job and could be encapsulded as an own class.
2. We should use namedtuples and dataclasses or some other kind of types to make the code better readable and easier to work with. We could do something like I did with the Operations and add a class for each major type of variable like Edges, Vertices, etc. Especially to avoid something like this `c[1][1]["avg_vol"]` or that `path[i][1]["stoptime"]` which I find incredibly hard to read and get through

Reply:
yeah for readability this would be a major improvement, but i don't know how named tuples and such impact performance, because the graph algo is what currently needs the majority of computing time.

What we should try is to save the timerange and volume data directly in the graph itself and not how i do it currently (using self.cache), using this the algorithm could be far more efficient, due to the ability of being able to filter the pairs while walking the paths, reducing the iterations significantly when timerangedata is present.

Reply:
It should be fine for us when we switch to dataclasses. My intuitive opinion on this is that we should prefer readability over performance. 
https://stackoverflow.com/a/55256047

We might want to consider the usage of `__slots__`. This could be interesting for that https://stackoverflow.com/questions/50180735/how-can-dataclasses-be-made-to-work-better-with-slots.